core/src/main/scala/kafka/server/KafkaBroker.scala

```diff
@@ -71,8 +70,11 @@ object KafkaBroker {
 }
 
 trait KafkaBroker extends KafkaMetricsGroup {
+  @volatile private var _brokerState: BrokerState = BrokerState.NOT_RUNNING
```

Comment:
Why not using ```@volatile var brokerState: BrokerState = BrokerState.NOT_RUNNING```?

Reply:
Because that would make it mutable for everyone. The current implementation is mutable only within subclasses.